### ISSUE TITLE
Keras train_on_batch not updating network

### ISSUE THREAD

**Issue by stylianosc07** [keras-team/keras#14581]
I am trying to train a GAN model and I am using train_on_batch to update the network but it seems that even though it calculates new losses it doesn't save the changes to the network. I tried printing losses before and after the training and the before and after are the same. What is the problem?
```
print(d_model_transfer.test_on_batch(Xsup_real, [ysup_real, label_sup_real]))
d_results = d_model_transfer.train_on_batch(Xsup_real, [ysup_real, label_sup_real])
print(d_results)
print(d_model_transfer.test_on_batch(Xsup_real, [ysup_real, label_sup_real]))
```
```
[0.08498648554086685, 0.0006484165205620229, 0.08433806896209717, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
[0.23932093381881714, 0.0010019546607509255, 0.23831897974014282, 1.0, 1.0, 1.0, 1.0, 0.875, 0.8749999403953552, 0.875, 0.875]

[0.08498648554086685, 0.0006484165205620229, 0.08433806896209717, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
```

**Comment by bguetarni** [comment]
What are these values ? Can you display the loss for the generator and the loss for the discriminator in a comprehensible way please ?

This can be due to an over-powerful discriminator that has quickly learned to discriminate between real and generated data. Thus, the generator is not learning anymore, because the discriminator provides no useful feedback for him.
If you provide your code in a notebook with the data, maybe we could investigate. Good luck :) !

**Comment by sushreebarsa** [comment]
@stylianosc07 In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by sushreebarsa** [comment]
Closing this issue due to lack of recent activity  .Please feel free to reopen if you still have concern.Thanks!
